src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1630,6 +1654,63 @@ private void MakeSeparatorList(ReadOnlySpan<char> separators, ref ValueListBuild
             }
         }
 
+        private void MakeSeparatorListVectorized(ref ValueListBuilder<int> sepListBuilder, char c, char? c2 = null, char? c3 = null)
+        {
+            // Constant that defines indices of characters within an AVX-Register
+            const ulong indicesConstant = 0xFEDCBA9876543210;
+            // Constant that allows for the truncation of 16-bit (FFFF/0000) values within a register to 4-bit (F/0)
+            Vector256<byte> shuffleConstant = Vector256.Create(0x02, 0x06, 0x0A, 0x0E, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
+                                                               0xFF, 0xFF, 0xFF, 0xFF, 0x02, 0x06, 0x0A, 0x0E, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF);
+
+            Vector256<ushort> v1 = Vector256.Create(c);
+            Vector256<ushort> v2 = c2 is char sep2 ? Vector256.Create(sep2) : v1;
+            Vector256<ushort> v3 = c3 is char sep3 ? Vector256.Create(sep3) : v2;
+
+            ref char c0 = ref MemoryMarshal.GetReference(this.AsSpan());
+            int cond = Length - (Length % Vector256<ushort>.Count);
```

Comment:
```suggestion
            int cond = Length & -Vector256<ushort>.Count;
```
Is the same, but as [bit-hack](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAJI8ovLrl5hcAbho1iAZiakGAYQYBvGg11Nl4jAwCyACkMMANjC4BzDAAsAlFp173xAOxWb9hwzgGU2s7RwYAUgYANRgwDGhSAFYkAB4OXAdoDAA+FjUIbgwnOWp3AF83XUr9BgtjcnMeH1DnV1L3PS9mvwYAMgCYuITktIys3PzCkvKaMqA===)